Update package metadata for Amazon.Lambda.Annotations (#980)
aws/aws-lambda-dotnet: Libraries/src/Amazon.Lambda.Annotations/README.md

# Amazon.Lambda.Annotations

> The Lambda Annotation is a programming model for writing .NET Lambda function. At a high level the programming model allows for 

Reviewer: ```suggestion
The Lambda Annotations is a programming model for writing .NET Lambda function. At a high level the programming model allows
```

> idiomatic .NET coding patterns and uses C# source generator technology to bridge the gap between the Lambda programming model 

Reviewer: ```suggestion
idiomatic .NET coding patterns and uses [C# Source Generators](https://docs.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview) to bridge the gap between the Lambda programming model 
```

{
    public void ConfigureServices(IServiceCollection services)
    {
        services.AddAWSService<Amazon.S3.IAmazonS3>();
        services.AddScoped<ITracker, DefaultTracker>();
    }
}
```

Example function using DI:
```csharp
public class Functions
{
    IAmazonS3 _s3Client;

    public Functions(IAmazonS3 s3Client)
    {
        _s3Client = s3Client;
    }

    [LambdaFunction]
    [HttpApi(HttpMethod.Put, HttpApiVersion.V2, "/process/{name}")]
    public async Task Process([FromServices] ITracker tracker, string name, [FromBody] string data)
    {
        tracker.Record();

        await _s3Client.PutObjectAsync(new PutObjectRequest
        {
            BucketName = "storage-bucket",
            Key = name,
            ContentBody = data
        });
    }
}
```

## Lambda .NET Attributes

List of .NET attributes currently supported.

* LambdaFunction
    * Placed on a method. Indicates this method should be exposed as a Lambda function.
* LambdaStartup
    * Placed on a class. Indicates this type should be used as the startup class and is used to configure the dependency injection and middleware. There can only be one class in a Lambda project with this attribute.

> ### Event Attributes    

Reviewer: Should include one liner for no event attached lambda function. They can also be written using Annotations and take the advantage of template generation, DI etc.

Author: Add a description to the event attribute section and mentioned that you don't have to have an event attribute.

* RestApi
    * Configures the Lambda function to be called from an API Gateway REST API. The HTTP method and resource path are required to be set on the attribute.
* HttpApi
    * Configures the Lambda function to be called from an API Gateway HTTP API. The HTTP method, HTTP API payload version and resource path are required to be set on the attribute.

### Parameter Attributes

* FromHeader
    * Map method parameter to HTTP header value
* FromQuery

>     * Map method parameter to query string pareamter

Reviewer: ```suggestion
    * Map method parameter to query string parameter
```

* FromRoute

>     * Map method parameter to resource path segement

Reviewer: ```suggestion
    * Map method parameter to resource path segment
```
